x-pack/plugin/security/src/test/java/org/elasticsearch/xpack/security/authc/AuthenticationServiceTests.java

```diff
@@ -297,17 +299,26 @@ public void testTokenMissing() throws Exception {
 
             Mockito.doReturn(List.of(secondRealm)).when(realms).getUnlicensedRealms();
             Mockito.doReturn(List.of(firstRealm)).when(realms).asList();
-            final String reqId = AuditUtil.getOrGenerateRequestId(threadContext);
+            boolean requestIdAlreadyPresent = randomBoolean();
+            SetOnce<String> reqId = new SetOnce<>();
+            if (requestIdAlreadyPresent) {
+                reqId.set(AuditUtil.getOrGenerateRequestId(threadContext));
+            }
```

Comment:
Tests in this class have been modified to ensure that a `requestId` is set when done.